taskcluster/ci/ui-test-apk/kind.yml

```diff
@@ -7,12 +7,13 @@ loader: taskgraph.loader.transform:loader
 transforms:
     - android_taskgraph.transforms.gradle_optimization:transforms
     - android_taskgraph.transforms.test:transforms
+    - android_taskgraph.transforms.slack_notify:transforms
+    - taskgraph.transforms.notify:transforms
```

Comment:
Ah, when we were chatting over zoom I didn't realize the test tasks themselves were doing the notification. That's totally fine, just a misunderstanding on my part.

In this case, to only show the notification on release graphs, you'd add something like:
```
if config.params.get("shipping_phase") not in ("ship", "promote"):
    yield task
    continue
    
# logic to add `notify` section
```

to your transform function. That way, it'll only notify when it runs on release graphs.

Reply:
I will just mention, that in case you wanted to have a single notification that runs after *all* test tasks have finished, then you'd want to have a dedicated "notify" task that depends on all the tasks that it should be notifying about.

Here's an example of one such task in the `mozilla-vpn-client` repo:
https://github.com/mozilla-mobile/mozilla-vpn-client/blob/main/taskcluster/kinds/release-notify/kind.yml

Reply:
i tried setting this up using
```
worker-type: succeed
worker:
    implementation: succeed
```
but i'm getting an assertion error from `set_implementation` because i guess `implementation` is already being set in the worker somewhere else...i'm guessing from the other `notify` tasks?

so i'm guessing i'm inheriting some stuff I don't know about?

also, in my new task, i'm using a python script, which needs some dependencies that might not be in the docker image used for `succeed`, but i can't even get that far to see if it works; so i've just been using a different work-type, the same as here, `b-android`.

some help here would be greatly appreciated.